### PR TITLE
AF-XXX: Added more unit tests for VimeoSessionManager

### PR DESCRIPTION
**Pull request by mikew-personal**
#### Ticket

N/A

#### Pull Request Checklist

- [x] Resolved any merge conflicts
- [x] No build errors or warnings are introduced
- [x] New files are written in Swift
- [x] New classes contain license headers
- [x] New classes have Documentation
- [x] New public methods have Documentation

#### Issue Summary

- Added two new unit tests to check functionality of VimeoSessionManager account handling methods.

#### Implementation Summary

- VimeoSessionManagerTests - added a unit test to verify that the session manager properly sets the accessTokenProvider on the request serializer from a VIMAccount and a test that verifies it's properly removed.

#### How to Test

- Run the unit tests

### CODE REVIEW

**Comment by mikew-personal** [comment]
build this please

**Comment by cameo-github** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/69504d79b74082ca47cc2215ad7509e350a606b3/VimeoNetworking/Sources/VimeoClient.swift#L330">VimeoNetworking/Sources/VimeoClient.swift#L330</a></strong>: cast from ‘Response&lt;VIMNullResponse&gt;’ to unrelated type ‘Response&lt;ModelType&gt;’ always fails  <br /><code>
                let response = Response(model: nullResponseObject, json: [:]) as! Response&lt;ModelType&gt;
</code></td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 48 tests, with 0 failures (0 unexpected) in 0.602 (0.633) seconds</td>
    </tr>
  </tbody>
</table>

See build details on [CircleCI](https://circleci.com/gh/vimeo/VimeoNetworking/390)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOSTests/VimeoSessionManagerTests.swift [hunk 1]
```diff
@@ -115,4 +114,55 @@ class VimeoSessionManagerTests: XCTestCase
...
+        sessionManager.clientDidAuthenticate(with: testAccount)
+        
+        guard let requestSerializer = sessionManager.requestSerializer as? VimeoRequestSerializer else
+        {
+            XCTFail("Incorrect request serializer")
+            return
+        }
+        
+        XCTAssertEqual(requestSerializer.accessTokenProvider?(), testAccount.accessToken)
+
```
Comment (jasonhawkins) [inline VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOSTests/VimeoSessionManagerTests.swift]: nit: extra white space hereee

**Comment by mikew-personal** [comment]
@jasonhawkins Definitely we should be writing unit tests to cover both expected behavior + failure handling code.  This PR was more about testing the codecov integration vs. writing out all the test coverage.  However, in this particular case there isn't much error handling to test.  Despite their names, these functions don't actually have anything to do with authentication.  They simply set or remove the accessToken block on the request serializer and return the accessToken that's passed in with the VIMAccount.  The actual authentication code which does have error cases would be in the AuthenticationController class.

**Pull request merged**